mutable/slice.go

```diff
@@ -0,0 +1,95 @@
+package mutable
+
+import "math/rand"
+
+// Filter iterates over elements of collection, returning an array of all elements predicate returns truthy for.
+// The function returns the modified slice, which may be shorter than the original if some elements were removed.
+// The order of elements in the original slice is preserved in the output.
+// Play:
+func Filter[T any](collection *[]T, predicate func(item T) bool) {
+	FilterI(collection, func(item T, index int) bool {
+		return predicate(item)
+	})
+}
+
+// Filter iterates over elements of collection, returning an array of all elements predicate returns truthy for.
+// The function returns the modified slice, which may be shorter than the original if some elements were removed.
+// The order of elements in the original slice is preserved in the output.
+// Play:
+func FilterI[T any](collection *[]T, predicate func(item T, index int) bool) {
+	j := 0
+	for i := range *collection {
+		if predicate((*collection)[i], i) {
+			(*collection)[j] = (*collection)[i]
```

Comment:
iiuc, this is `immutable`, so I think you should dump the filtered elems into a new collection, or ?

Reply:
nope ;)

this subpackage would be dedicated to mutable helpers

Reply:
ok, it is more safe to work on a copied variable, since the iteration might mutate the collection and this could cause subtle hidden bugs iiuc 

P.S. dont vote to the right :D

Reply:
I'm not sure to understand. You think that `predicate()` could manipulate `collection` ?

PS: don't worry 😅✊

Reply:
```
(*collection)[j] = (*collection)[i]
```